design/one-pager-merge-jets-proposal.md

```diff
@@ -0,0 +1,370 @@
+decides to start its own provider, then the story is still clear; there is a new
+provider maintained by its vendor and you should migrate to that one once you
+see fit as opposed to Jet vs classic choice where you need to think about
+different trade-offs with different and uncertain timelines.
+
+### Ownership by The Cloud Vendors
+
+As stated above, we hope that cloud vendors will come in and maintain their
+Crossplane providers. It seems like Terraform has achieved this goal only for
+GCP; Azure and AWS Terraform providers are not maintained by the vendors. In our
+case, Crossplane is a CNCF project with an open governance model, so we can't
+really compare it with Terraform, a property of HashiCorp, in that respect but
+it still provides a data point.
+
+The strategy we choose should take the following possibilities into account:
+* They may choose to bootstrap their own Crossplane provider, maybe even not
+under Crossplane organization.
+* They might decide to convert the underlying implementation to their own
+technology, similar to GCP Terraform provider that is [undergoing a
+process][gcp-tf-contributing] where they change implementation from handwritten
+API calls to Magic Modules and DCL.
+* They would provide the tools but not take part in maintenance of the codebase.
+
+In the first case, regardless of whether we go with separate or mixed providers,
+we will likely suggest users to migrate to the vendor-maintained provider so
+that they can get commercial grade support for the bugs in the provider. One
+could argue that if users are on a single provider once that happens, it's
+probably easier to handle the migration. In addition, it's likely the main
+reason they'd step in would be community-maintained provider having a large user base,
+so it's not very likely that they will start a new provider and handle migration
+from a provider with such a large user base.
+
+In the second case, mixed provider wouldn't require any manual migration. The
+conversion webhooks that Kubernetes API provides will take care of the API
+changes as described in [provider strategy doc][strategy-doc] in detail. In the
+case of separate providers though, users of one of the provider would have to
+migrate.
+
+In the third case, if we go with single mixed provider, we can choose to use the
+cloud vendor tooling for the resources we decide to convert. If we go with
+separate providers, we could keep working with those tools, like AWS CC, in the
+existing classic provider or bootstrap a new one but either case would either be
+similar or worse than the single mixed provider in terms of choices users would
+have to consider. Overall, the decision of how to go from Terrajet to
+cloud-vendor tooling is an independent decision from how we manage Terrajet and
+classic provider mix since it's the step after we decide that, if ever, the
+provided tooling is better than we have. For example, today, calling AWS API
+directly is definitely not better than Terrajet.
```

Comment:
> For example, today, calling AWS API directly is definitely not better than Terrajet.

Is this that much of a foregone conclusion? Is there something readers can reference to demonstrate it? I would presume that like anything there are trade-offs here. Off the top of my head:

* Provider maintainers can move faster without a dependency on Terraform (an upstream project).
* Invoking Terraform binaries is presumably more resource intensive than making an SDK call.

Reply:
I have updated this section to point to some of the trade-offs. Do you think having an explicit comparison would help?